### PR TITLE
fix(no_stopping_area): skip drawing a stop line in the case where the detected stop lines are goals

### PR DESCRIPTION
**Pull request by kaigohirao**
## Description

Related PR: https://github.com/autowarefoundation/autoware.universe/pull/6517
Related Issue: https://github.com/autowarefoundation/autoware.universe/issues/7081
Related JIRA Tickets: https://tier4.atlassian.net/browse/RT1-6419, https://tier4.atlassian.net/browse/RT1-5039

In the previous PR https://github.com/autowarefoundation/autoware.universe/pull/6517, I changed source code in the no stopping area module as a temporal solution 
to the problem reported and it resulted in the wrong implementation of the module as pointed out in the issue https://github.com/autowarefoundation/autoware.universe/issues/7081.

For the current PR, I reverted the change of pull/6517 and gave another solution to the original problem.

### Detail of the change to solve the problem of PR https://github.com/autowarefoundation/autoware.universe/pull/6517.

I consulted with @kosuke55 san and gave another solution to the problem of not being able to
enter into the no stopping area when the goal is right behind the no stopping areas.

In `checkStopLinesInNoStoppingArea` of  `scene_no_stopping_area_module.cpp`,
I inserted the lines of checking if the stop line detected is near 1m of goal or not.
If the first stop line detected inside the predicted path is near goal, then the no stopping area after this change
skips its own process and does not draw a stop line before the no stopping area.

## Tests performed

I confirmed the bug reported in JIRA Ticket https://tier4.atlassian.net/browse/RT1-5039 is resolved, using current version of Autoware.

2024/06/11 https://evaluation.tier4.jp/evaluation/reports/0f925436-b286-5c9d-b72d-6977817653c7/?project_id=prd_jt

## Effects on system behavior

Before this change and after the change in https://github.com/autowarefoundation/autoware.universe/pull/6517, 
No stopping area module had not distinguished the two cases of stop lines and stuck vehicles beyond the no stopping area,
and also had not took into account the module margin parameters, `stop_line_margin`, `stop_margin` and `stuck_vehicle_front_margin`. 

After this change, the module handles the three cases of goals, stop lines other than goals, and stuck vehicles beyond the no stopping area. These behavior for the latter two cases other than goals is compatible with the current [documentation](https://autowarefoundation.github.io/autoware.universe/main/planning/behavior_velocity_no_stopping_area_module/#no-stopping-area) for the no stopping area module.

## Interface changes

No Interface changed with this PR.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_velocity_no_stopping_area_module/src/scene_no_stopping_area.cpp [hunk 1]
```diff
@@ -251,6 +251,10 @@ bool NoStoppingAreaModule::checkStopLinesInNoStoppingArea(
   const tier4_planning_msgs::msg::PathWithLaneId & path, const Polygon2d & poly)
 {
   const double stop_vel = std::numeric_limits<float>::min();
+
+  // if the detected stop point is near goal, it's ignored.
+  const double close_to_goal_distance = 1.0;
```
Comment (shmpwk) [inline planning/behavior_velocity_no_stopping_area_module/src/scene_no_stopping_area.cpp:256]:
```suggestion
  constexpr double close_to_goal_distance = 1.0;
```

**Comment by kosuke55** [comment]
sorry to be late. I'm checking with tier4 internal scenario tests.
2024/06/11 https://evaluation.tier4.jp/evaluation/reports/0f925436-b286-5c9d-b72d-6977817653c7/?project_id=prd_jt

**Comment by kosuke55** [comment]
2024/06/13 https://evaluation.tier4.jp/evaluation/reports/ec205ca1-d9a8-5684-9fc2-c03041fa45fa/?project_id=prd_jt

**Comment by kosuke55** [comment]
https://evaluation.tier4.jp/evaluation/oidc/auth/cb/?code=ory_ac_j1iHna8do4GgdvcRbydEebRyTMQAfJQDJddpJ4mt22I.EWv7zP5FOfJxDfz7RC6jCTHc3KkaLyphrzUBWAogEEM&scope=openid+profile+email+offline_access&state=f9f8cdd63ad344ee9642405dc3965132

**Review by kosuke55 — APPROVED** [review]
LGTM

**Pull request merged**